### PR TITLE
bamql 1.4

### PR DESCRIPTION
**Pull request by tseemann**
- [ ] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by sjackman** [comment]
CircleCI on macOS
```
checking for  uuid ... no
configure: error: Package requirements ( uuid ) were not met:

No package 'uuid' found
```
https://circleci.com/gh/brewsci/homebrew-bio/1522

Perhaps `ossp-uuid` will work. Odd that this error doesn't occur on Linux.

**Comment by sjackman** [comment]
@apmasell Any thoughts on this build error on macOS, Andre? Does `bamql` require `uuid`, and for what purpose?
See the failed build log at https://circleci.com/gh/brewsci/homebrew-bio/1522

**Comment by apmasell** [comment]
A UUID is placed in the header of each output BAM so that accept/reject file pairs can be verified. `ossp-uuid` is the correct package in Brew on MacOS. It does not occur on Linux because `libuuid` required by `e2fsprogs` and `NetworkManager`.

**Comment by sjackman** [comment]
Thanks for the explanation, Andre.
@tseemann This should fix it:
```ruby
depends_on "ossp-uuid" if OS.mac?
```

**Comment by tseemann** [comment]
@sjackman `libuuid` is everywhere... i think Linux worked because I included `util-linux`. I didn't  know about `oss-uuid`. 

```
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /home/linuxbrew/.linuxbrew
Could not symlink share/man/man3/uuid.3
Target /home/linuxbrew/.linuxbrew/share/man/man3/uuid.3
is a symlink belonging to util-linux. You can unlink it:
  brew unlink util-linux
```

The problem is the *clashes* :-(

I'll assume Linux ppl will get it from util-linux and mac ppl from ossp-uuid....

```
brew info libuuid
libuuid: stable 1.0.3 (bottled)
  ossp-uuid (because both install lib/libuuid.a)
  util-linux (because both install lib/libuuid.a)
```

**Comment by tseemann** [comment]
Seems to not be compatible with `ossp-uuid` ?

```
/usr/local/Cellar/ossp-uuid/1.6.2_2/include/ossp/uuid.h:97:52: note: passing argument to parameter '_uuid' here
extern uuid_rc_t     uuid_create   (      uuid_t **_uuid);
                                                   ^
: error: use of undeclared identifier 'uuid_unparse'
    uuid_unparse(uuid, id_buf);
    ^
tools/run.cpp:49:5: error: use of undeclared identifier 'uuid_unparse'
    uuid_unparse(uuid, id_buf);
    ^
4 errors generated.
```

**Comment by apmasell** [comment]
Yes, the API is different.

**Comment by apmasell** [comment]
I've added support to this in head. I have no way to test it under `ossp-uuid` since I do not have a MacOS environment and the documentation links for the library are broken, so if it doesn't build, please send a PR.

**Comment by sjackman** [comment]
Should this formula be `depends_on :linux` until the next release that supports `ossp-uuid`?

**Comment by apmasell** [comment]
I'm not going to release a new version until this is resolved as there's been no other feature improvements.

**Comment by sjackman** [comment]
@apmasell Are you agreeing that `bamql` does not currently build on macOS, and should be `depends_on :linux`?

**Comment by apmasell** [comment]
No. I'm saying someone who has a Mac should download the bamql git repository and test that it builds on MacOS and submit a patch if it doesn't. It doesn't make sense for me to cut a new release where the only substantial change is adding code that doesn't build.

**Comment by sjackman** [comment]
Thanks for the clarification. I'll test it.

**Comment by sjackman** [comment]
```ruby
depends_on "llvm@4"
```
@tseemann Why does `bamql` depend specifically on `llvm@4`? Does it need this dependency on both Linux and macOS?

**Comment by apmasell** [comment]
BAMQL can be compiled with clang or GCC, but it still needs LLVM as a dependency. BAMQL *is* a compiler, so it uses LLVM to generate assembly code. clang uses LLVM too, so it might be possible to use the system LLVM on MacOS assuming Apple installed all the build dependencies.

**Comment by sjackman** [comment]
The clang provided by Apple does not include the LLVM libraries needed by BAMQL.
I see this commit `Add LLVM 5.0 support` https://github.com/BoutrosLaboratory/bamql/commit/6c6e5eecb4a991c56824be57ecc7de95921f832f. Is it released? Do you plan to add support for LLVM 6? Compiling HEAD with LLVM 6 gives this error.

```
tools/compile.cpp:41:10: fatal error: 'llvm/Target/TargetSubtargetInfo.h' file not found
#include <llvm/Target/TargetSubtargetInfo.h>
```

**Comment by sjackman** [comment]
I've verified that `bamql` HEAD works with `llvm@5` and `ossp-uuid`.
@apmasell Could you please tag a stable release?

**Comment by apmasell** [comment]
I updated to support LLVM 6. The new version is 1.5.

**Comment by sjackman** [comment]
Great. Thanks, Andre!

File: Formula/bamql.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  sha256 "92d1c88811d4d90df7864d2c3493d3378534c74a581378aeb352dd37365c5d09"
+  head "https://github.com/BoutrosLaboratory/bamql.git"
+
+  depends_on "autoconf"   => :build
+  depends_on "automake"   => :build
+  depends_on "libtool"    => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "htslib"
+  depends_on "llvm@4"
```
[the remark this answers is not in the document]
Comment (sjackman) [reply Formula/bamql.rb]: The new version of `bamql` can use the current `llvm` version 6.

**Comment by sjackman** [comment]
@tseemann Would you like to take a look at updating this PR to `bamql` 1.5, which can use `llvm` rather than `llvm@4` ?

**Pull request closed without merging**